core/src/main/scala/org/apache/spark/sql/snappyParsers.scala

```diff
@@ -630,14 +633,21 @@ class SnappyParser(caseSensitive: Boolean)
  * Snappy dialect uses a much more optimized parser and adds SnappyParser
```

Comment:
Would a thread local not work here ? Sometimes deep down inside rules it becomes cumbersome to access a SnappyContext.